sklearn/manifold/_spectral_embedding.py

```diff
@@ -366,6 +369,7 @@ def spectral_embedding(
             # approximation X to eigenvectors
             X = random_state.rand(laplacian.shape[0], n_components + 1)
             X[:, 0] = dd.ravel()
+            X = X.astype(laplacian.dtype)
```

Comment:
I wish it were possible to specify dtype when using numpy PRNG!

Reply:
I thought the same. But we need to discuss the SLEP about the random state